src/dev/renovate/package_groups.ts

```diff
@@ -0,0 +1,159 @@
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { pkg } from '../../legacy/utils/package_json';
+import { wordRegExp } from './utils';
+
+interface PackageGroup {
+  /**
+   * The group name, will be used for the branch name and in pr titles
+   */
+  readonly name: string;
+
+  /**
+   * Specific words that, when found in the package name, identify it as part of this group
+   */
+  readonly packageWords?: string[];
+
+  /**
+   * Exact package names that should be included in this group
+   */
+  readonly packageNames?: string[];
+
+  /**
+   * Extra labels to apply to PRs created for packages in this group
+   */
+  readonly extraLabels?: string[];
+}
+
+export const RENOVATE_PACKAGE_GROUPS: PackageGroup[] = [
+  {
+    name: 'eslint',
+    packageWords: ['eslint'],
+  },
+
+  {
+    name: 'babel',
+    packageWords: ['babel'],
+    packageNames: ['core-js'],
+  },
+
+  {
+    name: 'jest',
+    packageWords: ['jest'],
+  },
+
+  {
+    name: 'mocha',
+    packageWords: ['mocha'],
+  },
+
+  {
+    name: 'karma',
+    packageWords: ['karma'],
+  },
+
+  {
+    name: 'gulp',
+    packageWords: ['gulp'],
+  },
+
+  {
+    name: 'grunt',
+    packageWords: ['grunt'],
+  },
+
+  {
+    name: 'angular',
+    packageWords: ['angular'],
+  },
+
+  {
+    name: 'd3',
+    packageWords: ['d3'],
+  },
+
+  {
+    name: 'react',
+    packageWords: ['react', 'redux', 'enzyme'],
+    packageNames: ['ngreact', 'recompose', 'prop-types', 'typescript-fsa-reducers', 'reselect'],
+  },
+
+  {
+    name: 'graphql',
+    packageWords: ['graphql'],
+  },
+
+  {
+    name: 'webpack',
+    packageWords: ['webpack', 'loader'],
+    packageNames: ['mini-css-extract-plugin', 'chokidar'],
+  },
+
+  {
+    name: 'language server',
+    packageNames: ['vscode-jsonrpc', 'vscode-languageserver', 'vscode-languageserver-types'],
+  },
+
+  {
+    name: 'hapi',
+    packageWords: ['hapi'],
+    packageNames: ['hapi', 'joi', 'boom', 'hoek', 'h2o2', '@elastic/good', 'good-squeeze', 'inert'],
+  },
+
+  {
+    name: 'dragselect',
+    packageNames: ['dragselect'],
+    extraLabels: [':ml'],
+  },
+];
+
+/**
+ * Auto-define package groups for any `@types/*` deps that are not already in a group
+ */
+for (const dep of [...Object.keys(pkg.dependencies), ...Object.keys(pkg.devDependencies)]) {
```

Comment:
so it will handle only type packages from `package.json`. [others](https://github.com/elastic/kibana/pull/38369/files#diff-fb098e156f6841cfad609154ad7c309eR28) aren't included. are they?

Reply:
Oh, right, maybe we should add the other packages too 😄